src/UI/Component/Item/Item.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Item;
+
+/**
+ * Common interface to all items.
+ */
+interface Item extends \ILIAS\UI\Component\Component {
+	/**
+	 * Gets the title of the item 
+	 *
+	 * @return string
+	 */
+	public function getTitle();
+
+	/**
+	 * Create a new item with an attached description.
+	 * @param string $description
+	 * @return AppointmentItem
+	 */
+	public function withDescription($description);
+
+	/**
+	 * Get the description of the item.
+	 * @return string
+	 */
+	public function getDescription();
+
+	/**
+	 * Get a new item with the given properties as key-value pairs.
+	 *
+	 * The key is holding the title and the value is holding the content of the
+     * specific data set.
+	 *
+	 * @param array<string,string> $properties Title => Content
+	 * @return self
+	 */
+	public function withProperties(array $properties);
```

Comment:
I agree, but somehow I am reluctant with these add methods. Its always possible to do call get...(), extend the array and with...() again. If we withAdditional... I would certainly prefer to be able to add multiple items with one call.